diff --git a/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand.dtso b/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand.dtso
index fded878..272420d 100644
--- a/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand.dtso
+++ b/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand.dtso
@@ -65,6 +65,11 @@
 				#address-cells = <1>;
 				#size-cells = <1>;
 				reg = <0>;
+      	nand-ecc-mode = "soft";
+      	nand-ecc-algo = "bch";
+      	nand-ecc-strength = <4>;
+      	nand-ecc-step-size = <512>;
+      	nand-is-boot-medium;
 
 				spi-max-frequency = <52000000>;
 				spi-tx-bus-width = <4>;
