diff --git a/net/alist/Makefile b/net/alist/Makefile
index 9b0cf2f..9c3ee6c 100644
--- a/net/alist/Makefile
+++ b/net/alist/Makefile
@@ -64,6 +64,11 @@ define Build/Prepare
 	)
 endef
 
+define Build/Compile
+	$(call GoPackage/Build/Compile)
+	$(STAGING_DIR_HOST)/bin/upx --lzma --best $(GO_PKG_BUILD_BIN_DIR)/alist
+endef
+
 ifneq ($(CONFIG_USE_MUSL),)
   TARGET_CFLAGS += -D_LARGEFILE64_SOURCE
 endif
